---
title: "otu_json"
author: "Niklas Rindtorff"
date: "12/9/2018"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```

I load required packages. 

```{r}
library(phyloseq)
library(curatedMetagenomicData)
```

I download all microbiome datasets publicly available from the curated metagenomics.

```{r, message=FALSE, eval=FALSE}
retagenomicData()[grep("metaphlan_bugs_list.stool",curatedMetagenomicData())]
data = curatedMetagenomicData(x=list_datasets,dryrun=FALSE)
save(data,file="microbiome_data.RData")
```

```{r}
load("microbiome_data.RData")
```

I convert the standardized experiment format into a phyloseq object.

```{r}
pseq = data %>% 
  mergeData(.) %>% 
  ExpressionSet2phyloseq(.)
```



```{r}
pseq
```


```{r}
#df <- otu_table(pseq)[1:20, 1]
```

```{r}
df <- otu_table(pseq)[, 1]
```


```{r}
library(jsonlite)
library(mirobiomefhir)
obj_syn <- fromJSON("../microbiome_synthesis.json")
#save(obj_syn, file = "fhir_template.RData")
template <- obj_syn$contained
template$id <- df %>% as.data.frame() %>% rownames()


tidy_obs <- function(df){
  lubridate::ymd_hms(template$issued)
}
```

```{r}
test <- create_obs(snomed_id = NA, species_id = df %>% rownames() %>% as.vector() %>%.[1], percent_abundance = df[,1] %>% as.vector() %>% .[1])
```

```{r}
tmp <- df %>% as.data.frame() %>% 
  magrittr::set_colnames("abundance") %>% 
  rownames_to_column("id")

a <- as.list(tmp$abundance)
b <- as.list(tmp$id)

res_df <- create_res(tmp$id)
obs_df <- map2(a,b, ~ create_obs(percent_abundance = a, species_id = b))
obs_df <- base::mapply(create_obs, percent_abundance = tmp$abundance, species_id = tmp$id, SIMPLIFY = FALSE)


library(data.table)

#obs_df_bind <- obs_df %>% rbindlist()
```


```{r}
obj_syn <- fromJSON("../microbiome_synthesis.json")
test <- obj_syn

test$contained <- df_tmp
test$result <- res_df
```


```{r}
test %>% toJSON(dataframe = "rows", pretty = TRUE) %>% write("test_bind.json")
```

